File: programmers/sorting/biggest_number.py
from functools import cmp_to_key

def comp(a, b):
    if a + b <= b + a:
        return 1
    else:
        return -1

def solution(numbers):
    a = [str(x) for x in numbers]
    a.sort(key=cmp_to_key(comp))

    answer = ''
    for c in a:
        answer += c
    
    if answer[0] == '0':
        answer = '0'

    return answer

File: programmers/sorting/biggest_number.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {        
        ArrayList<String> num = new ArrayList<>();
        for (int x: numbers) {
            num.add(String.valueOf(x));
        }
        
        Collections.sort(num, (s1, s2) -> {
            StringBuilder sb1 = new StringBuilder(s1);
            sb1.append(s2);
            StringBuilder sb2 = new StringBuilder(s2);
            sb2.append(s1);
            return sb2.compareTo(sb1);
        });
        
        StringBuilder res = new StringBuilder();
        for (String x: num) {
            res.append(x);
        }
        
        String answer = res.toString();
        if (answer.charAt(0) == '0') answer = "0";
        
        return answer;
    }
}